include/rmm/mr/device/device_memory_resource.hpp

```diff
@@ -155,6 +157,164 @@ class device_memory_resource {
     return do_is_equal(other);
   }
 
+  /**
+   * @brief Allocates memory of size at least \p bytes.
+   *
+   * The returned pointer will have at minimum 256 byte alignment.
+   *
+   * If supported, this operation may optionally be executed on a stream.
+   * Otherwise, the stream is ignored and the null stream is used.
```

Comment:
suggestion: I think this doc is stale or copy/pasted from elsewhere since this function doesn't take a stream. This applies throughout this header.